### DISCUSSION TITLE
useSharedValue array update

### DISCUSSION THREAD

**Discussion by lukasa1993** [software-mansion/react-native-reanimated#3592] [Q&A]
https://gist.github.com/lukasa1993/9310d1fae54cb540edcf390305f6f2f6

any way to optimize line 50 that line basically does setState i wonder if there is a way to use pathList without state update

**Comment by tomekzaw** [comment]
Hey, check out `useAnimatedProps` hook that allows you to set native properties of components from worklets. The only limitation here is that you cannot create new or remove existing views from worklets - they need to be managed by React. I would recommend refactoring your code to eliminate `_pathList` array and use a single `AnimatedPath` instead with `useAnimatedProps`.
